### DISCUSSION TITLE
Addition of Text Functionality in Albums for Adding Context and Narratives

### DISCUSSION THREAD

**Discussion by axgd-code** [immich-app/immich#3896] [Feature Request]
# Description
Currently, the use of albums in the project only allows storing and viewing images, without the ability to add contextual information or associated narratives. I would like to propose adding a feature that enables users to input and display text within albums, similar to what is possible with Google Photos.

## Use Cases:
- A user creates an album for a trip. effectively narrating the progression of the journey day by day.
- (bonus) Dynamic Number of Fields: The number of generated text fields should be based on the range of dates within the album, ensuring accuracy and relevance.

## Context:
When creating albums for events such as trips, celebrations, or other special occasions, it's often important to provide contextual details and narratives to accompany the images. This text functionality would allow users to create visual travel logs, where they can share their experiences and anecdotes.

# Key Features:

- Text Addition: Users should be able to add text in the form of descriptions, annotations, or narratives in an album.
- Text Display: The added text should be prominently visible beween the images in the album, allowing viewers to understand the context and story associated with each image.
- Basic Formatting: Provide basic text formatting options such as bold, italics, etc., to allow visually appealing presentation.
- Custom Order: Enable users to choose the display order of images and text to create a coherent narrative.
- Editing: Allow users to edit or update the text after initially adding it, providing ongoing flexibility to tell their story.

### Platform

- [ ] Server
- [X] Web
- [X] Mobile

**Comment by bo0tzz** [comment]
Relevant previous discussion on Discord: https://discord.com/channels/979116623879368755/1134549197379616859

**Comment by bo0tzz** [comment]
Related: #2208

**Comment by jwnacnud** [comment]
Would something like this do what you are looking for in Immich?  

https://memoryties.com/event/dbe13edb-cd7f-4cc4-89d9-74d18207397d

It allows stories and narratives and perspectives to be captured, both an audio and text and then stored along with the single image or any number of images in an album.

**Comment by aleqx** [comment]
Yes, please! Couples with children would be all over it (I would).

@jwnacnud how exactly does that work, if you don't mind? It seems it's an external website whereas immich would be self-hosted? Can it be self-hosted? If yes, would it imply that one needs to upload on immich, then open another website/app, find/link the photos, then add narratives? If yes, that's probably too much hassle for most

**Reply by jwnacnud** [reply]
From my perspective the story is the most important thing.  Sometimes there are stories *without* a picture.  Sometimes there are stories that have many pictures.  In most cases an image or video serves as a placeholder for the story/memory.

A picture is worth a thousand words, they say.  I say... but only if you write them down (or capture them somehow).

I build memoryties for stories and then added the ability to gather the supporting images along with the story/memory.

Since Immich is an API it would be possible to store the images in immich and then the stories in memoryties - serving the images from immich.  Memoryties is not self hosted (though that's an interesting idea).

Perhaps, within immich, a user could upload their image and then add a narrative that is stored/hosted within memoryties, and the narrative could be viewed alongside the image within Immich?

**Reply by aleqx** [reply]
Sounds like a bit too much hassle, both usage and maintainance

**Reply by SamBouwer** [reply]
@jwnacnud memoryties looks great! is it possible to self-host or is the source code available somewhere?

**Reply by jwnacnud** [reply]
@SamBouwer I have never considered exposing the source code or allowing it to be self-hosted, though that's an interesting idea.  What did you like about it?

**Reply by SamBouwer** [reply]
I love the combination of simplicity (serves one function and does it well), and the fact that it seems to allow for both typed and voice input, transcription of voice input into text, and playback of (I think) the original audio. Integrating this into Immich would be amazing for storing both the picture and the corresponding memories. I noticed it even might have user attribution to hook into Immich user accounts to see who contributed what memory.

**Reply by bo0tzz** [reply]
This discussion should probably move elsewhere, it's getting pretty off-topic.

**Reply by jwnacnud** [reply]
@bo0tzz Ack.

**Comment by Novapixel1010** [comment]
Thank you [bo0tzz](https://github.com/bo0tzz) for pointing me to the existing discussions. My apologies for the duplicate request; I wasn't able to find some of these older threads during my initial search.

On a related note, I think it would be extremely helpful to have a dedicated section, label, or pinned discussion/issue that consolidates all current feature requests related specifically to 'Albums.'

This would not only make it easier for users to find existing requests but would also help the development team track and prioritize all album-related features in one central place. 

**List of Feature Requests**
- #24542 same #18868: Markdown support
- #9040 Same #24543 
- #2208
- Similar to this request #15619
-  journaling and better/more metadata for albums #30317

**Reply by plans-coding** [reply]
Great compilation @Novapixel1010. Can you add mine too #30317 ?

**Comment by elmimmo** [comment]
I requested something like the audio track in @jwnacnud's link to [memoryties](https://memoryties.com/event/dbe13edb-cd7f-4cc4-89d9-74d18207397d), and naturally having it also as searchable text (as in that example) would be awesome:

• **[[Feature] Audio commentaries for pictures/albums](https://github.com/immich-app/immich/discussions/16646)**

It sounds to me like the most precious treasure to have people’s narration of pictures and albums along with them, specially recordings of people that passed away or that used to be little kids.

Alas, the reviewer did not share my enthusiasm and deemed it not something the dev team would look into.
